mono/sgen/sgen-gc.c

```diff
@@ -326,6 +326,15 @@ nursery_canaries_enabled (void)
 
 #define safe_object_get_size	sgen_safe_object_get_size
 
+#if defined(TARGET_OSX) || defined(TARGET_WIN32) || (defined(__linux__) && !defined(TARGET_ANDROID))
```

Comment:
I think better macros to use here would be `PLATFORM_MACOSX`, `HOST_WIN32`, `PLATFORM_LINUX`, and `PLATFORM_ANDROID`.